### PR TITLE
Cortex auth

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Description
Supports auth with demistobot using encrypted keys

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Additional changes
Describe additional changes done, for example adding a function to common server.

### CODE REVIEW

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py [hunk 1]
```diff
@@ -6,15 +6,25 @@
...
+import base64
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+
 # disable insecure warnings
 requests.packages.urllib3.disable_warnings()
 
 ''' GLOBAL VARS '''
 
-DEMISTO_APP_TOKEN = demisto.params().get('token')
+# Authorization token
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py]: remove these comments

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py [hunk 2]
```diff
@@ -52,6 +62,43 @@
...
+        # Create key
+        aes_gcm = AESGCM(enc_key)
+        # Create nonce
+        nonce = create_nonce()
+        # Create ciphered data
+        data = string.encode()
+        ct = aes_gcm.encrypt(nonce, data, None)
+        return base64.b64encode(nonce + ct)
+    now = epoch_seconds()
+    return encrypt(f'{now}:{auth_id}', key).decode("utf-8")
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py]: `"` -> `'`

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py [hunk 3]
```diff
@@ -52,6 +62,43 @@
 ''' HELPER FUNCTIONS '''
 
 
+def get_encrypted(auth_id: str, key: str):
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py]: add return type

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py [hunk 4]
```diff
@@ -110,21 +157,21 @@ def epoch_seconds(d=None):
...
-        'Authorization': DEMISTO_APP_TOKEN,
+        'Authorization': TOKEN,
         'Accept': 'application/json'
     }
+
     dbot_response = requests.get(
         TOKEN_RETRIEVAL_URL,
         headers=headers,
-        params={'token': DEMISTO_APP_TOKEN},
+        params={"token": get_encrypted(AUTH_ID, ENC_KEY)},
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py]: `'`

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml [hunk 5]
```diff
@@ -1,56 +1,54 @@
+category: Analytics & SIEM
 commonfields:
   id: Palo Alto Networks Cortex
   version: -1
-name: Palo Alto Networks Cortex
-fromversion: 4.1.0
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml]: add that back

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex_test.py [hunk 6]
```diff
@@ -80,3 +97,10 @@ def test_prepare_fetch_query(mocker):
     assert firewall_query_with_severity_and_subtype == "SELECT * FROM panw.threat WHERE receive_time>1524383011 " \
                                                        "AND (subtype='url' OR subtype='antivirus') " \
                                                        "AND (severity='medium')"
+
+
+def test_get_encrypted():
+    from PaloAltoNetworksCortex import get_encrypted
+    auth_id = random_string(50)
+    auth_key = random_string(32)
+    get_encrypted(auth_id, auth_key)
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex_test.py]: need to check that the actual value is the expected, so lets not do it with random string

Comment (jochman) [reply Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex_test.py]: the nonce (random int) is always random (we should never use the same nonce twice) so we don't know what will go out

Comment (jochman) [reply Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex_test.py]: I can add a decryption function to the test.

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml [hunk 7]
```diff
@@ -397,34 +538,49 @@ script:
...
+      type: Unknown
     - contextPath: Cortex.Logging.name-of-threatid
       description: The name of the threat ID.
       type: string
     - contextPath: Cortex.Logging.severity
       description: Threat Severity.
-    description: Runs a query on the Cortex logging service, according to preset queries.
-  dockerimage: demisto/python_pancloud:1.0.0.170
+      type: Unknown
+  dockerimage: devtesting/python_pancloud:1.0.0.1557754040
```
Comment (Itay4) [inline Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml]: update docker image

**Comment by Itay4** [comment]
pending https://github.com/demisto/dockerfiles/pull/60

File: Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml (hunk 7 above)
Comment (Itay4) [reply Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.yml]: @jochman update to `demisto/python_pancloud:1.0.0.286`

**Pull request merged**